tabular/src/autogluon/tabular/models/tab_transformer/tab_transformer_model.py

```diff
@@ -0,0 +1,474 @@
+""" TabTransformer model """
+import logging
+import os
+import time
+
+import numpy as np
+import pandas as pd
+from autogluon.core.utils.loaders import load_pkl
+from tqdm import tqdm
+
+from .hyperparameters.parameters import get_default_param
+from .hyperparameters.searchspaces import get_default_searchspace
+from ..abstract.abstract_model import AbstractModel
+from ...constants import BINARY, REGRESSION, MULTICLASS
+from autogluon.core.utils import try_import_torch
+
+logger = logging.getLogger(__name__)
+
+
+class TabTransformerModel(AbstractModel):
+    """
+    Main TabTransformer model that inherits from AbstractModel.
+
+    TabTransformer uses modifications to the typical Transformer architecture and the pretraining in BERT
+    and applies them to the use case of tabular data. Specifically, this makes TabTransformer suitable for unsupervised
+    training of Tabular data with a subsequent fine-tuning step on labeled data.
+    """
+    params_file_name = "tab_trans_params.pth"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.types_of_features = None
+        self.verbosity = None
+        self.temp_file_name = "tab_trans_temp.pth"
+        try_import_torch()
+
+    def _set_default_params(self):
```

Comment:
this is called before try_import_torch()

Reply:
Moved try_import_torch() to top of __init__